### PR TITLE
community: add some milvus async functions

### PR DESCRIPTION
**Pull request by nlp-dev-id**
**Description**: Add some async functions to milvus module: aadd_texts, asimilarity_search_with_score_by_vector, amax_marginal_relevance_search_by_vector
**Issue:**: Async wrappers for milvus i/o are not yet implemented
**Twitter handle:** niklas.kobelev@ippen-digital.de

### CODE REVIEW

File: libs/community/langchain_community/vectorstores/milvus.py [hunk 1]
```diff
@@ -613,6 +615,144 @@ def add_texts(
...
+                Defaults to 1000.
+            ids (Optional[List[str]]): List of text ids. The length of each item
+
+        Raises:
+            MilvusException: Failure to add texts
+
+        Returns:
+            List[str]: The resulting keys for each inserted element.
+        """
+        from pymilvus import Collection, MilvusException
```
Comment (isahers1) [inline libs/community/langchain_community/vectorstores/milvus.py:655]: imports should be wrapped in try/except

File: libs/community/langchain_community/vectorstores/milvus.py [hunk 2]
```diff
@@ -613,6 +615,144 @@ def add_texts(
...
+        metadatas: Optional[List[dict]] = None,
+        timeout: Optional[float] = None,
+        batch_size: int = 1000,
+        *,
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        """Insert text data into Milvus asynchronously.
+
+        Inserting data when the collection has not be made yet will result
```
Comment (isahers1) [inline libs/community/langchain_community/vectorstores/milvus.py:630]: typo: "be made" - "been made"

File: libs/community/langchain_community/vectorstores/milvus.py [hunk 3]
```diff
@@ -613,6 +615,144 @@ def add_texts(
...
+        if not self.auto_id:
+            assert isinstance(
+                ids, list
+            ), "A list of valid ids are required when auto_id is False."
+            assert len(set(ids)) == len(
+                texts
+            ), "Different lengths of texts and unique ids are provided."
+            assert all(
+                len(x.encode()) <= 65_535 for x in ids
+            ), "Each id should be a string less than 65535 bytes."
```
Comment (isahers1) [inline libs/community/langchain_community/vectorstores/milvus.py:667]: asserts can cause errors - I think wrapping in try/except is best way to go

File: libs/community/langchain_community/vectorstores/milvus.py [hunk 4]
```diff
@@ -779,6 +919,74 @@ def similarity_search_with_score_by_vector(
...
+                limit=k,
+                expr=expr,
+                output_fields=output_fields,
+                timeout=timeout,
+                _async=True,
+                **kwargs,
+            )
+            return search_future.result()
+
+        loop = asyncio.get_event_loop()
```
Comment (isahers1) [inline libs/community/langchain_community/vectorstores/milvus.py:976]: Is it guaranteed that an event loop is running?

Comment (isahers1) [reply libs/community/langchain_community/vectorstores/milvus.py:976]: Line 1188 too

**Comment by oubeichen** [comment]
Any updates on this issue?

**Comment by zc277584121** [comment]
Thank for the contribution, currently partner's code is move to libs/partner/milvus/langchain_milvus.  The old langchain_community package is outdated

**Pull request closed without merging**